inertia/inertia.go

```diff
@@ -6,50 +6,47 @@ import (
 	"net/http"
 )
 
-var (
-	rootTmpl *template.Template
+type inertiaCtxKeyType string
 
-	version     string
-	versionFunc func() string
-)
+const inertiaCtxKey inertiaCtxKeyType = "inertia"
 
-type page struct {
-	Component string      `json:"component"`
-	Props     interface{} `json:"props"`
-	URL       string      `json:"url"`
-	Version   string      `json:"version"`
-}
+type Inertia struct {
+	RootTemplate *template.Template
 
-func Init(rootTemplate *template.Template) {
-	rootTmpl = rootTemplate
+	Version     string
+	VersionFunc func() string
 }
 
-func Version(v string) {
-	version = v
+func (i *Inertia) GetVersion() string {
+	if i.VersionFunc != nil {
+		return i.VersionFunc()
+	}
+	return i.Version
 }
 
-func VersionFunc(f func() string) {
-	versionFunc = f
+type page struct {
+	Component string      `json:"component"`
+	Props     interface{} `json:"props"`
+	URL       string      `json:"url"`
+	Version   string      `json:"version"`
 }
 
-func GetVersion() string {
-	if versionFunc != nil {
-		return versionFunc()
+func Render(w http.ResponseWriter, r *http.Request, componentName string, props interface{}) {
+	inertia, ok := r.Context().Value(inertiaCtxKey).(*Inertia)
+	if !ok {
+		panic("[Inertia] No middleware configured.")
 	}
-	return version
-}
 
-func Render(w http.ResponseWriter, r *http.Request, componentName string, props interface{}) {
 	page := page{
 		Component: componentName,
 		Props:     props,
 		URL:       r.RequestURI,
-		Version:   GetVersion(),
+		Version:   inertia.GetVersion(),
 	}
 
 	marshalled, err := json.Marshal(page)
```

Comment:
for testability of the actual render logic, it would be nicer to have to deal with the context passing. So move the actual render logic to a function on the inertia instance. This will also be needed in the future when implementing share etc. 
```suggestion
	func (inertiaInstance *Interia) render(w ...) 
```

Reply:
Good idea!